File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/view/activity/base/GGHFRVModelActivity.java
package io.ganguo.demo.view.activity.base;

import io.ganguo.viewmodel.common.HFRecyclerViewModel;
import io.ganguo.viewmodel.common.HeaderItemViewModel;
import io.ganguo.viewmodel.common.HeaderViewModel;
import io.ganguo.viewmodel.databinding.IncludeHfRecyclerBinding;
import io.ganguo.vmodel.BaseViewModel;
import io.ganguo.vmodel.ViewModelHelper;

/**
 * HFRecyclerViewModel -- Activity
 * Created by leo on 2018/10/27.
 */
public abstract class GGHFRVModelActivity<T extends IncludeHfRecyclerBinding, B extends HFRecyclerViewModel> extends GGVModelActivity<T, B> {
    @Override
    public void onViewAttached(B viewModel) {
        ViewModelHelper.bind(viewModel.getHeaderContainer(), viewModel, onCreateHeaderVModel());
    }


}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/view/activity/base/GGHFRVModelActivity.kt
package io.ganguo.demo.view.activity.base

import io.ganguo.viewmodel.core.ViewModelHelper
import io.ganguo.viewmodel.databinding.IncludeHfRecyclerBinding
import io.ganguo.viewmodel.pack.common.HFRecyclerViewModel

/**
 * HFRecyclerViewModel -- Activity
 * Created by leo on 2018/10/27.
 */
abstract class GGHFRVModelActivity<T : IncludeHfRecyclerBinding, B : HFRecyclerViewModel<*>> : GGVModelActivity<T, B>() {
    override fun onViewAttached(viewModel: B) {
        ViewModelHelper.bind(viewModel.headerContainer, viewModel, newHeaderVModel())
    }

}
